moneta-core/src/main/java/org/javamoney/moneta/internal/ConfigurableCurrencyUnitProvider.java

```diff
@@ -69,19 +73,33 @@
             }
             return result;
         }
+        if (!currencyQuery.getNumericCodes().isEmpty()) {
+            for (Integer numericCode : currencyQuery.getNumericCodes()) {
+                CurrencyUnit cu = CURRENCY_UNITS_BY_NUMERIC_CODE.get(numericCode);
+                if (cu != null) {
+                    result.add(cu);
+                }
+            }
+            return result;
+        }
         result.addAll(CURRENCY_UNITS.values());
         return result;
     }
 
     /**
-     * Registers a bew currency unit under its currency code.
+     * Registers a bew currency unit under its currency code and potentially numeric code.
```

Comment:
bew->new

Reply:
Fixed it, will likely cause a conflict with #312.